test/mjsunit/simd/float32x4.js

```diff
@@ -942,3 +942,22 @@ function testArrayOfFloat32x4() {
 }
 
 testArrayOfFloat32x4();
+
+function testFloat32x4OveralignedLoad() {
+  var b = new ArrayBuffer(40);
+  var a = new Float32Array(b, 8);
+  var af = new Float64Array(b, 8);
+  for (var i = 0; i < a.length; i++) {
+    a[i] = i;
+  }
+
+  for (var i = 0; i < a.length - 3; i += 2) {
+    var v = SIMD.float32x4.load(af, i / 2);
```

Comment:
What's your indention to use `i` to compare with `a.length` and use it to load from `af`? Can we write some simpler test case for the byteOffset?